vivisect/base.py

```diff
@@ -669,19 +673,35 @@ def _initFunction(self, funcva):
     #def _loadImportApi(self, apidict):
         #self._imp_api.update( apidict )
 
+    def getEndian(self):
+        return self.bigend
+
+    def _doSetEndian(self, endian):
+        self.bigend = endian
+        for arch in self.imem_archs:
+            if not arch:
+                continue
+            arch.setEndian(self.bigend)
+
 
 #################################################################
 #
 #  setMeta key callbacks
 #
     def _mcb_Architecture(self, name, value):
+        archid = envi.getArchByName(value)
+        try:
+            # Some of the ENVI archs defined may not architecture modules that 
+            # are still in progress
+            self.setMemArchitecture(archid)
+        except IndexError:
+            raise Exception("Architecture Module not defined for %s yet!" % value)
```

Comment:
Minor nit: maybe we make a specific exception for this?

And also tweak this line slightly to `raise Exception(...) from None)`

Reply:
done